Chatter/Chatter.cs

```diff
@@ -336,6 +336,13 @@ class MessageHudPatch {
           return;
         }
 
+        if (!DisallowChatMessageHudStartTexts.Value.IsNullOrWhiteSpace()) {
```

Comment:
Leaving feedback before closing this PR as this would not have been merged due to the following:

First, this entire block is inefficient in that it will constantly parse the config value on every SystemHud-type message (which can be many due to Hud spam) even if the config value has not changed.

So multiple config value entries should be cached in a collection and only updated on a BepInExConfiguation.SettingChanged event. Additionally, this creates a new array every single time due to Split() leading to unnecessary garbage accumulation when a HashSet<string> would've avoided that.

Second, using 'comma' as a split value for entries completely eliminates values that have any form of text that has a comma in it. Which is an artificial restriction that can be worked around using another form of values storage.

Third ties in with the first, where disallowedString.Trim() would be part of updating any cached values. Also, this is **not** indicated at all in the ConfigHelp message and there may be some use cases where they would also want to have a space at the end/start of a string for filtering.

Fourth, this only handles filtering for "startWith" and does not allow for "endWith" or "contains" configuration which may be of interest to some users. I do see that this config option is named specifically for StartTexts and as an initial implementation would be fine, but it would be best to design for flexibility.

Fifth, This does not conform to the code style I have chosen to use for all my mods. I have not included a C# linter since I do formatting manually and well no one else contributes to this repo but myself. For this particular section, all if statements must have {} and return must be on it's own separate line.

I hope I'm conveying these comments as constructive feedback as I only extensively write this much in reviews I do for work all for the purposes of educating and improving code quality and readability.

Reply:
I understand. Thanks for pointing out things I need to improve stuff on my end. Looking forward to your Chatter updates.